### ISSUE TITLE
Cannot refresh list of messages

### ISSUE THREAD

**Issue by livthomas** [dhis2msg/DHISmessenger#17]
If you try to refresh Messages list, you will get _Something went wrong. Try to refresh_. I can see this exception in the Android Studio:

```
org.json.JSONException: No value for read
    at org.json.JSONObject.get(JSONObject.java:355)
    at org.json.JSONObject.getString(JSONObject.java:515)
    at org.dhis2.messenger.gui.fragment.InboxFragment$4.doInBackground(InboxFragment.java:415)
    at org.dhis2.messenger.gui.fragment.InboxFragment$4.doInBackground(InboxFragment.java:347)
    at android.os.AsyncTask$2.call(AsyncTask.java:288)
    at java.util.concurrent.FutureTask.run(FutureTask.java:237)
    at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
    at java.lang.Thread.run(Thread.java:841)
```

I am able to reproduce this when I send a message from user **_android**_ through web interface to user **_livthomas**_ using the app.

@MathiasCiarlo @jiaderc Do you have any idea where the problem can be?

**Comment by VladislavAlfredov** [comment]
Strange... it works fine with my local server.
I just checked the server version... it is a newer build.
"Version:
    2.21
Build revision:
    20886
Build date:
    2015-11-12 12:23"
Whilst my local one is :
"Version:
    2.21
Build revision:
    20840
Build date:
    2015-10-23 02:03"

My best guess is that they are changing the api (?) 
So instead of reading the "read" field, let's just add it as is, and the cache will set the read/not read if the newer object's date is newer than the old one.
I will commit a solution in a few...

**Comment by VladislavAlfredov** [comment]
Ok I have implemented the fix (095cfa7bc284e59a3ace20ccd223b5bc49e03ca2).
As such this bug is closed. :)
